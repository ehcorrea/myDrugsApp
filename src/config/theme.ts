export default {
  border: {
    radius: {
      xsmall: 4,
      small: 8,
      medium: 10,
      large: 30,
    },
  },
  font: {
    family: {
      title: 'PlayfairDisplay-Regular',
      titleItalic: 'PlayfairDisplay-Italic',
      italic: 'Poppins-Italic',
      medium: 'Poppins-Medium',
      mediumItalic: 'Poppins-MediumItalic',
      regular: 'Poppins-Regular',
      semiBold: 'Poppins-SemiBold',
      semiBoldItalic: 'Poppins-SemiBoldItalic',
    },
    wight: {
      light: 300,
      normal: 400,
      bold: 600,
      xbold: 700,
    },
    sizes: {
      xsmall: 12,
      small: 14,
      medium: 16,
      large: 20,
      xlarge: 24,
      xxlarge: 28,
      huge: 32,
    },
  },
  colors: {
    primary: '#F231A5',
    secondary: '#3CD3C1',
    mainBg: '#EFEFEF',
    lightBg: '#06092B',
    white: '#FAFAFA',
    black: '#000000',
    lightGray: '#E1E1E1',
    gray: '#545454',
    darkGray: '#2E2F42',
    red: '#A74141',
  },
  spacings: {
    xxsmall: 8,
    xsmall: 14,
    medium: 16,
    large: 20,
    xlarge: 28,
    xxlarge: 32,
    huge: 44,
  },
} as const;
