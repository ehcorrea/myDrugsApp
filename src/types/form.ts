import { RefAttributes } from 'react';
import { TextInput } from 'react-native';
import { ControllerRenderProps } from 'react-hook-form';

export type FormComponentProps = {
  hasError: boolean;
  onChangeText: ControllerRenderProps['onChange'];
} & Pick<ControllerRenderProps, 'value'> &
  RefAttributes<TextInput>;
