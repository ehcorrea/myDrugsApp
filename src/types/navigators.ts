export type PublicNavigatorParamList = {
  InitialScreen: undefined;
  LoginScreen: undefined;
  SingUpScreen: undefined;
};

export type NavigationParamList = PublicNavigatorParamList;
