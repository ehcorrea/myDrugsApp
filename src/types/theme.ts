import theme from '@/config/theme';

export type Color = keyof typeof theme.colors;
export type FontFamily = keyof typeof theme.font.family;
export type FontSize = keyof typeof theme.font.sizes;
