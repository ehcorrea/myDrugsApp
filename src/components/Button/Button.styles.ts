import styled, { css } from 'styled-components/native';

import { BorderRadius } from '@/types/theme';

export type ContainerProps = {
  rounded?: BorderRadius;
};

export const Container = styled.TouchableOpacity<ContainerProps>`
  ${({ theme, rounded }) => css`
    align-items: center;
    background-color: ${theme.colors.black};
    height: ${theme.spacings.huge}px;
    justify-content: center;
    width: 222px;

    ${rounded &&
    css`
      border-radius: ${theme.border.radius[rounded]}px;
    `}
  `}
`;
