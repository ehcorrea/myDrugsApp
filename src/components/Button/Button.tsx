import React from 'react';
import { TouchableOpacityProps, TouchableOpacity } from 'react-native';

import { Text } from '../Text/Text';

import * as S from './Button.styles';

export type ButtonProps = {
  children?: React.ReactNode;
  label?: string;
} & TouchableOpacityProps &
  S.ContainerProps;

export function Button({ children, label, ...props }: ButtonProps) {
  const content = label ? <Text color="white">{label}</Text> : children;
  const accessibilityLabel = props.accessibilityLabel || label;

  return (
    <S.Container accessibilityLabel={accessibilityLabel} {...props}>
      {content}
    </S.Container>
  );
}

type ButtonLinkProps = {
  label: string;
} & Omit<ButtonProps, 'children'>;

Button.Link = ({ label, ...props }: ButtonLinkProps) => {
  return (
    <TouchableOpacity {...props}>
      <S.Label>{label}</S.Label>
    </TouchableOpacity>
  );
};
