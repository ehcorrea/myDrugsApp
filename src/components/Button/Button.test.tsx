import { Text } from 'react-native';

import { fireEvent, render } from '@/test/utils';
import { BorderRadius } from '@/types/theme';
import theme from '@/config/theme';

import { Button } from './Button';

const onPress = jest.fn();

test('[label] success render Button', () => {
  const container = render(<Button onPress={onPress} label="button label" />);
  const button = container.getByText('button label');

  fireEvent.press(button);
  expect(onPress).toHaveBeenCalled();

  expect(container).toMatchSnapshot();
});

test('[children] success render Button', () => {
  const container = render(
    <Button onPress={onPress}>
      <Text>button label</Text>
    </Button>
  );
  const button = container.getByText('button label');

  fireEvent.press(button);
  expect(onPress).toHaveBeenCalled();

  expect(container).toMatchSnapshot();
});

test.each(Object.keys(theme.border.radius))(
  '[border %s] success render Button',
  (value) => {
    const border = value as BorderRadius;
    const container = render(<Button label="button label" rounded={border} />);

    const button = container.getByLabelText('button label');
    expect(button).toHaveStyle({ borderRadius: theme.border.radius[border] });

    expect(container).toMatchSnapshot();
  }
);

test('should success render Button.Link', () => {
  const container = render(
    <Button.Link label="button label" onPress={onPress} />
  );
  const button = container.getByText('button label');

  fireEvent.press(button);
  expect(onPress).toHaveBeenCalled();

  expect(container).toMatchSnapshot();
});
