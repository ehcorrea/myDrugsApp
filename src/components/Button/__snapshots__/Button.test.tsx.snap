// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[border large] success render Button 1`] = `
<View
  accessibilityLabel="button label"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "borderRadius": 20,
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "width": 222,
    }
  }
>
  <Text
    color="white"
    style={
      [
        {
          "color": "#FAFAFA",
          "fontFamily": "Poppins-Regular",
          "fontSize": 31,
        },
      ]
    }
  >
    button label
  </Text>
</View>
`;

exports[`[border medium] success render Button 1`] = `
<View
  accessibilityLabel="button label"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "borderRadius": 10,
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "width": 222,
    }
  }
>
  <Text
    color="white"
    style={
      [
        {
          "color": "#FAFAFA",
          "fontFamily": "Poppins-Regular",
          "fontSize": 31,
        },
      ]
    }
  >
    button label
  </Text>
</View>
`;

exports[`[border small] success render Button 1`] = `
<View
  accessibilityLabel="button label"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "borderRadius": 8,
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "width": 222,
    }
  }
>
  <Text
    color="white"
    style={
      [
        {
          "color": "#FAFAFA",
          "fontFamily": "Poppins-Regular",
          "fontSize": 31,
        },
      ]
    }
  >
    button label
  </Text>
</View>
`;

exports[`[border xsmall] success render Button 1`] = `
<View
  accessibilityLabel="button label"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "borderRadius": 4,
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "width": 222,
    }
  }
>
  <Text
    color="white"
    style={
      [
        {
          "color": "#FAFAFA",
          "fontFamily": "Poppins-Regular",
          "fontSize": 31,
        },
      ]
    }
  >
    button label
  </Text>
</View>
`;

exports[`[children] success render Button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "width": 222,
    }
  }
>
  <Text>
    button label
  </Text>
</View>
`;

exports[`[label] success render Button 1`] = `
<View
  accessibilityLabel="button label"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "width": 222,
    }
  }
>
  <Text
    color="white"
    style={
      [
        {
          "color": "#FAFAFA",
          "fontFamily": "Poppins-Regular",
          "fontSize": 31,
        },
      ]
    }
  >
    button label
  </Text>
</View>
`;

exports[`should success render Button.Link 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    color="red"
    size="small"
    style={
      [
        {
          "color": "#A74141",
          "fontFamily": "Poppins-Regular",
          "fontSize": 27,
        },
        {
          "textTransform": "capitalize",
        },
      ]
    }
  >
    button label
  </Text>
</View>
`;
